controllers/grafanamutetiming_controller.go

```diff
@@ -0,0 +1,252 @@
+	return ctrl.Result{RequeueAfter: muteTiming.Spec.ResyncPeriod.Duration}, nil
+}
+
+func (r *GrafanaMuteTimingReconciler) reconcileWithInstance(ctx context.Context, instance *grafanav1beta1.Grafana, muteTiming *grafanav1beta1.GrafanaMuteTiming) error {
+	cl, err := client2.NewGeneratedGrafanaClient(ctx, r.Client, instance)
+	if err != nil {
+		return fmt.Errorf("building grafana client: %w", err)
+	}
+
+	_, err = r.getMuteTimingByName(ctx, muteTiming.Spec.Name, instance)
+	shouldCreate := false
+	if err != nil {
+		if errors.Is(err, provisioning.NewGetMuteTimingNotFound()) {
+			shouldCreate = true
+		} else {
+			return fmt.Errorf("getting mute timing by name: %w", err)
+		}
+	}
+
+	trueRef := "true" //nolint:goconst
+	editable := true
+	if muteTiming.Spec.Editable != nil && !*muteTiming.Spec.Editable {
+		editable = false
+	}
+
+	var payload models.MuteTimeInterval
+	payload.Name = muteTiming.Spec.Name
+	payload.TimeIntervals = make([]*models.TimeIntervalItem, 0, len(muteTiming.Spec.TimeIntervals))
+	for _, ti := range muteTiming.Spec.TimeIntervals {
+		times := make([]*models.TimeIntervalTimeRange, 0, len(ti.Times))
+		for _, tr := range ti.Times {
+			times = append(times, &models.TimeIntervalTimeRange{
+				StartTime: tr.StartTime,
+				EndTime:   tr.EndTime,
+			})
+		}
+		payload.TimeIntervals = append(payload.TimeIntervals, &models.TimeIntervalItem{
+			DaysOfMonth: ti.DaysOfMonth,
+			Location:    ti.Location,
+			Months:      ti.Months,
+			Weekdays:    ti.Weekdays,
+			Years:       ti.Years,
+			Times:       times,
+		})
+	}
+	if shouldCreate {
+		params := provisioning.NewPostMuteTimingParams().WithBody(&payload)
+		if editable {
+			params.SetXDisableProvenance(&trueRef)
+		}
+		_, err = cl.Provisioning.PostMuteTiming(params) //nolint:errcheck
+		if err != nil {
+			return fmt.Errorf("creating mute timing: %w", err)
+		}
+	} else {
+		params := provisioning.NewPutMuteTimingParams().WithName(muteTiming.Spec.Name).WithBody(&payload)
+		if editable {
+			params.SetXDisableProvenance(&trueRef)
+		}
+		_, err = cl.Provisioning.PutMuteTiming(params) //nolint:errcheck
+		if err != nil {
+			return fmt.Errorf("updating mute timing: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *GrafanaMuteTimingReconciler) getMuteTimingByName(ctx context.Context, name string, instance *grafanav1beta1.Grafana) (*models.MuteTimeInterval, error) {
+	cl, err := client2.NewGeneratedGrafanaClient(ctx, r.Client, instance)
+	if err != nil {
+		return nil, fmt.Errorf("building grafana client: %w", err)
+	}
+
+	muteTiming, err := cl.Provisioning.GetMuteTiming(name)
+	if err != nil {
+		return nil, fmt.Errorf("getting mute timing: %w", err)
+	}
+
+	return muteTiming.Payload, nil
+}
+
+func (r *GrafanaMuteTimingReconciler) finalize(ctx context.Context, muteTiming *grafanav1beta1.GrafanaMuteTiming) error {
+	r.Log.Info("Finalizing GrafanaMuteTiming")
+
+	instances, err := GetAllMatchingInstances(ctx, r.Client, muteTiming)
```

Comment:
We discussed what the expected behaviour was on finalize, and it was decided to make `allowCrossNamespaceImport` partially immutable
```suggestion
	instances, err := GetScopedMatchingInstances(r.Log, ctx, r.Client, muteTiming)
```